microbench/src/test/java/io/netty/microbench/util/AbstractExecutorMicrobenchmark.java

```diff
@@ -0,0 +1,136 @@
+import io.netty.util.concurrent.Promise;
+
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.Fork;
+
+/**
+ * This harness facilitates the sharing of an executor between JMH and Netty and
+ * thus avoid measuring context switching in microbenchmarks.
+ */
+@Fork(AbstractExecutorMicrobenchmark.DEFAULT_FORKS)
+public class AbstractExecutorMicrobenchmark extends AbstractMicrobenchmarkBase {
+
+    protected static final int DEFAULT_FORKS = 0; // Forks has to be 0 so tasks are run immediately by JMH
+
+    public static void executor(EventLoop service) {
+        DelegateHarnessExecutor.executor(service);
+    }
+
+    public static final class DelegateHarnessExecutor extends AbstractEventExecutor {
+        private static EventLoop executor;
+        public DelegateHarnessExecutor(int maxThreads, String prefix) {
+            System.out.println("Using ImmediateHarnessExecutor executor " + this);
+        }
+
+        public static void executor(EventLoop service) {
+            executor = service;
+        }
+
+        @Override
+        public boolean inEventLoop() {
+            return executor.inEventLoop();
+        }
+
+        @Override
+        public boolean inEventLoop(Thread thread) {
+            return executor.inEventLoop(thread);
+        }
+
+        @Override
+        public Future<?> shutdownGracefully(long quietPeriod, long timeout, TimeUnit unit) {
+            return executor.shutdownGracefully(quietPeriod, timeout, unit);
+        }
+
+        @Override
+        public Future<?> terminationFuture() {
+            return executor.terminationFuture();
+        }
+
+        @Override
+        @Deprecated
+        public void shutdown() {
+            executor.shutdown();
+        }
+
+        @Override
+        public boolean isShuttingDown() {
+            return executor.isShuttingDown();
+        }
+
+        @Override
+        public boolean isShutdown() {
+            return executor.isShutdown();
+        }
+
+        @Override
+        public boolean isTerminated() {
+            return executor.isTerminated();
+        }
+
+        @Override
+        public boolean awaitTermination(long timeout, TimeUnit unit) {
+            try {
+                return executor.awaitTermination(timeout, unit);
+            } catch (InterruptedException e) {
+                handleUnexpectedException(e);
+            }
+            return false;
+        }
+
+        @Override
+        public void execute(Runnable command) {
+            executor.execute(command);
+        }
+
+        @Override
+        public <V> Promise<V> newPromise() {
+            return executor.newPromise();
+        }
+
+        @Override
+        public <V> ProgressivePromise<V> newProgressivePromise() {
+            return executor.newProgressivePromise();
+        }
+    }
+
+    protected static final String[] JVM_ARGS = {
+        "-server", "-dsa", "-da", "-ea:io.netty...", "-Xms2g", "-Xmx2g",
```

Comment:
.. scratch that, I see you have to overwrite. Consider creating a BASE_JVM_ARGS array